### PR TITLE
doc: Add API reference docs

### PR DESCRIPTION
**Pull request by duwenxin99**
1. Set up API reference documents
2. Update github workflow to test for documentation correctness with nox
3. Add change log, html template, and convert README file from .md to .rst
4. add requirements.txt in docker/docs

### CODE REVIEW

**Comment by jackwotherspoon** [comment]
@duwenxin99 do you mind updating the PR description quickly summarizing this PR for future proofing. Also I think this PR should use the `docs:` prefix instead of `feat:` as feat should be saved for features of the package/library itself.

File: docs/README.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+../README.rst
```
Comment (averikitsch) [inline docs/README.rst:1]: Use symlink

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
+AlloyDB for PostgreSQL for LangChain
```
Comment (averikitsch) [inline README.rst:1]: Can we match the exact format in https://github.com/googleapis/langchain-google-cloud-sql-pg-python/blob/main/README.rst?plain=1

Comment (duwenxin99) [reply README.rst:1]: matched

File: docs/README.rst (hunk 1 above)
Comment (duwenxin99) [reply docs/README.rst:1]: added symlink

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+==================================================
+
+- |preview| |pypi| |versions|
+
+- `Client Library Documentation`_
+- `Product Documentation`_
+- `API Reference`_
```
Comment (averikitsch) [inline README.rst:8]: Remove API reference. Use API reference link for Client Library Documentation instead

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+steps:
+
+1. `Select or create a Cloud Platform project.`_
+2. `Enable billing for your project.`_
+3. `Enable the Google Cloud AlloyDB API.`_
+4. `Setup Authentication.`_
+
+.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
+.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
+.. _Enable the Google Cloud AlloyDB API.: https://console.cloud.google.com/flows/enableapi?apiid=alloydb.googleapis.com
```
Comment (averikitsch) [inline README.rst:34]:
```suggestion
.. _Enable the AlloyDB API.: https://console.cloud.google.com/flows/enableapi?apiid=alloydb.googleapis.com
```

API names can be found https://cloud.google.com/guides/style/branding-list

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+Use a vector store to store embedded data and perform vector search.
+
+.. code-block:: python
+
+   from langchain_google_alloydb_pg import AlloyDBEngine, AlloyDBVectorStore
+   from langchain_google_vertexai import VertexAIEmbeddings
+
+
+   engine = AlloyDBEngine.from_instance("project-id", "region", "my-cluster", "my-instance", "my-database")
+   embeddings_service = VertexAIEmbeddings()
```
Comment (averikitsch) [inline README.rst:95]:
```suggestion
   embeddings_service = VertexAIEmbeddings(model_name="textembedding-gecko@003")
```
I forgot to update the readme previously with code changes.

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+   vectorstore = AlloyDBVectorStore(
+       engine,
+       table_name="my-table",
+       embeddings=embedding_service
```
Comment (averikitsch) [inline README.rst:99]:
```suggestion
   vectorstore = AlloyDBVectorStore.create_sync(
       engine,
       table_name="my-table",
       embedding_service=embedding_service
```

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+Use a document loader to load data as LangChain ``Document``\ s.
+
+.. code-block:: python
+
+   from langchain_google_alloydb_pg import AlloyDBEngine, AlloyDBLoader
+
+
+   engine = AlloyDBEngine.from_instance("project-id", "region", "my-cluster", "my-instance", "my-database")
+   loader = PostgresSQLLoader(
```
Comment (averikitsch) [inline README.rst:117]:
```suggestion
   loader = AlloyDBLoader.create_sync(
```

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+Use ``ChatMessageHistory`` to store messages and provide conversation
+history to LLMs.
+
+.. code:: python
+
+   from langchain_google_alloydb_pg import AlloyDBChatMessageHistory, AlloyDBEngine
+
+
+   engine = AlloyDBEngine.from_instance("project-id", "region", "my-cluster", "my-instance", "my-database")
+   history = AlloyDBChatMessageHistory(
```
Comment (averikitsch) [inline README.rst:139]:
```suggestion
   history = AlloyDBChatMessageHistory.create_sync(
```

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,174 @@
+AlloyDB for PostgreSQL for LangChain
+==================================================
+
+- |preview| |pypi| |versions|
```
Comment (averikitsch) [inline README.rst:4]:
```suggestion
|preview| |pypi| |versions|
```

**Review by averikitsch — CHANGES_REQUESTED** [review]

**Review by averikitsch — APPROVED** [review]

**Pull request merged**